cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetDiff.java

```diff
@@ -107,7 +108,7 @@ public StatefulSetDiff(StatefulSet current, StatefulSet desired) {
             // Any volume claim template changes apart from size change should trigger rolling update
             // Size changes should not trigger rolling update. Therefore we need to separate these two in the diff.
             changesVolumeClaimTemplate |= equalsOrPrefix("/spec/volumeClaimTemplates", pathValue) && !VOLUME_SIZE.matcher(pathValue).matches();
-            changesVolumeSize |= VOLUME_SIZE.matcher(pathValue).matches();
+            changesVolumeSize |= !isVolumeSizeEqual(pathValue, lookupPath(source, pathValue), lookupPath(target, pathValue));
```

Comment:
I think you should try to optimize the code a bit. You should probably first check the path whether it matches the VOLUME_SIZE path. Only then you should do the lookups.

If you keep using separate method for it, it migth also make more sense to have a method `isVolumeSizeChanged` instead of using ``isvolumeSizeEqual`. It will make the code more readable.